datadog_lambda/wrapper.py

```diff
@@ -105,24 +118,59 @@ def __call__(self, event, context, **kwargs):
 
     def _before(self, event, context):
         try:
+
             set_cold_start()
             submit_invocations_metric(context)
             # Extract Datadog trace context from incoming requests
-            extract_dd_trace_context(event)
+            dd_context = extract_dd_trace_context(event)
+
+            self.span = None
+            if dd_tracing_enabled:
+                self.span = self._create_dd_trace_py_span(context, dd_context)
+            else:
+                set_correlation_ids()
 
-            # Set log correlation ids using extracted trace context
-            set_correlation_ids()
             logger.debug("datadog_lambda_wrapper _before() done")
         except Exception:
             traceback.print_exc()
 
     def _after(self, event, context):
         try:
+            if self.span:
+                self.span.finish()
             if not self.flush_to_log:
                 lambda_stats.flush(float("inf"))
             logger.debug("datadog_lambda_wrapper _after() done")
         except Exception:
             traceback.print_exc()
 
+    def _create_dd_trace_py_span(self, context, trace_context):
+        span_context = None
+        if (
+            trace_context["source"] == TraceContextSource.EVENT
+            or self.merge_xray_traces
+        ):
+            headers = get_dd_trace_context()
+            span_context = self.propagator.extract(headers)
+
+        tags = {}
+        if context:
+            tags = {
+                "cold_start": is_cold_start(),
+                "function_arn": context.invoked_function_arn,
+                "request_id": context.aws_request_id,
+                "resource_names": context.function_name,
+            }
+        args = {
+            "service": self.function_name,
```

Comment:
Let's default service name to `aws.lambda` for future proof. Going to look a little bit odd (starting a new service) if merge_xray_traces is true, but I think that' probably fine, and user should be able to override the service name using a tag to combine the two xray spans and this dd span under one service.

Reply:
Should we do this now, if we don't yet support the service tag override? Also, this won't appear on the trace view yet, since it searches by the service name, not the resource name.